students/chelsea_nayan/lesson03/tests/test_unit.py

```diff
@@ -0,0 +1,206 @@
+                      '(206)111-1111', 'aanacortes@email.com', 'Inactive', 10.00),
+                     ('02', 'Bilbo', 'Baggins', '200 2nd Ave E', '(206)222-222', # Customer [1]
+                      'bbaggins@email.com', 'Active', 20500.00),
+                     ('03', 'Charlie', 'Chadmeister', '300 3rd Ave S', # Customer [2]
+                      '(206)333-3333', 'cchadmesiter@gmail.com', 'Active', 4000),
+                     ('04', 'Danny', 'Devito', '400 4th Ave W', '(206)444-444', # Customer [3]
+                      'ddevito@email.com', 'Inactive', 500000)]
+    LOGGER.info('Intialized ')
+
+    def test_add_customer(self):
+        '''Testing add_customer function'''
+        LOGGER.info('Testing add_customer function!')
+        setup()
+
+        add_customer(*self.CUSTOMER_LIST[0])
+        add_customer(*self.CUSTOMER_LIST[1])
+        add_customer(*self.CUSTOMER_LIST[2])
+        add_customer(*self.CUSTOMER_LIST[3])
+
+        test_customer_01 = Customer.get(Customer.c_id == '01')
+        test_customer_02 = Customer.get(Customer.c_id == '02')
+        test_customer_03 = Customer.get(Customer.c_id == '03')
+        test_customer_04 = Customer.get(Customer.c_id == '04')
+
+        # Testing customer 01
+        self.assertEqual(test_customer_01.c_firstname, self.CUSTOMER_LIST[0][1])
+        self.assertEqual(test_customer_01.c_lastname, self.CUSTOMER_LIST[0][2])
+
+        # Testing customer 02
+        self.assertEqual(test_customer_02.c_home_address, self.CUSTOMER_LIST[1][3])
+        self.assertEqual(test_customer_02.c_phone_number, self.CUSTOMER_LIST[1][4])
+
+        # Testing customer 03
+        self.assertEqual(test_customer_03.c_email_address, self.CUSTOMER_LIST[2][5])
+        self.assertEqual(test_customer_03.c_status, self.CUSTOMER_LIST[2][6])
+
+        # Testing customer 04
+        self.assertEqual(test_customer_04.c_id, self.CUSTOMER_LIST[3][0])
+        self.assertEqual(test_customer_04.c_firstname, self.CUSTOMER_LIST[3][1])
+        self.assertEqual(test_customer_04.c_lastname, self.CUSTOMER_LIST[3][2])
+        self.assertEqual(test_customer_04.c_home_address, self.CUSTOMER_LIST[3][3])
+        self.assertEqual(test_customer_04.c_phone_number, self.CUSTOMER_LIST[3][4])
+        self.assertEqual(test_customer_04.c_email_address, self.CUSTOMER_LIST[3][5])
+        self.assertEqual(test_customer_04.c_status, self.CUSTOMER_LIST[3][6])
+        self.assertEqual(test_customer_04.c_credit_limit, self.CUSTOMER_LIST[3][7])
+
+        with self.assertRaises(IntegrityError):
+            add_customer(*self.CUSTOMER_LIST[6])
+
+        LOGGER.info('Finished testing add_customer function!')
+
+    def test_search_customer(self):
+        '''Testing search_customer function'''
+
+        LOGGER.info('Testing search_customer function!')
+        setup()
+
+        add_customer(*self.CUSTOMER_LIST[3]) # Danny Devito
+        expected = {'customer_id': '04',
```

Comment:
The expected return does not match the format used by your code for returning customer data as a dictionary:

```
        return {'firstname': some_customer.c_firstname,
                'lastname': some_customer.c_lastname,
                'email_address': some_customer.c_email_address,
                'phone_number': some_customer.c_phone_number}
```

You're including _customer_id_, _home_address_, _status_ and _credit_line_, none of which are being returned by _search_customer()_ in _basic_operations.py_, which means the test will always fail.